### PR TITLE
Add a new request parameter in Produce API to allow clients to choose …

### PR DESCRIPTION
**Pull request by cancecen**
…where they get ACKs for the messages they produce.

Client: Client gets the acks and the response is not sent until Proxy gets ACKs from all replicas.
Proxy: Proxy gets the acks and does not forward it to the client. Client request succeeds, as long as the request to *proxy* is successful.

### CODE REVIEW

File: src/main/java/server/KafkaProxyServiceImpl.java [hunk 1]
```diff
@@ -69,15 +70,21 @@ public void produce(
...
-        producerWrapper.produce(
-            request.getMessage().getMessageKey().getBytes(),
-            request.getMessage().getMessageContent().getBytes());
-    final ProduceResponse response =
-        ProduceResponse.newBuilder()
-            .setResponseCode(ResponseCode.OK)
-            .setOffset(kafkaResponse.offset())
-            .build();
+    final ProduceResponse response;
+    final byte[] key = request.getMessage().getMessageKey().getBytes();
```
Comment (afalko) [inline src/main/java/server/KafkaProxyServiceImpl.java]: Possible to have null message? null key?

File: src/main/java/server/KafkaProxyServiceImpl.java [hunk 2]
```diff
@@ -69,15 +70,21 @@ public void produce(
...
-    final RecordMetadata kafkaResponse =
-        producerWrapper.produce(
-            request.getMessage().getMessageKey().getBytes(),
-            request.getMessage().getMessageContent().getBytes());
-    final ProduceResponse response =
-        ProduceResponse.newBuilder()
-            .setResponseCode(ResponseCode.OK)
-            .setOffset(kafkaResponse.offset())
-            .build();
+    final ProduceResponse response;
```
Comment (afalko) [inline src/main/java/server/KafkaProxyServiceImpl.java]: Why clientpool non-final https://github.com/cancecen/kafkagrpcproxy/pull/7/files#diff-0d3ad750ed1e32f2b6da756cb5a3091dR32 ?

File: src/main/java/server/examples/AckedClient.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import org.kafkagrpcproxy.RegisterProducerResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AckedClient {
+
+  private static final Logger logger = LoggerFactory.getLogger(Client.class);
+
+  public static void main(String[] args) {
+    ManagedChannel channel =
```
Comment (afalko) [inline src/main/java/server/examples/AckedClient.java]: recommend putting this in try () {} resources block and take out the channel.close

File: src/main/java/server/KafkaProxyServiceImpl.java (hunk 1 above)
Comment (cancecen) [reply src/main/java/server/KafkaProxyServiceImpl.java:74]: good point. will fix.

File: src/main/java/server/KafkaProxyServiceImpl.java (hunk 2 above)
Comment (cancecen) [reply src/main/java/server/KafkaProxyServiceImpl.java:73]: I will make it final

File: src/main/java/server/examples/AckedClient.java (hunk 3 above)
Comment (cancecen) [reply src/main/java/server/examples/AckedClient.java:25]: sounds good

**Pull request merged**